pyanaconda/modules/storage/bootloader/utils.py

```diff
@@ -164,3 +165,62 @@ def install_boot_loader(storage):
 
     # Install the bootloader.
     storage.bootloader.write()
+
+
+def recreate_initrds(sysroot, kernel_versions):
+    """Recreate the initrds by calling new-kernel-pkg or dracut.
+
+    This needs to be done after all configuration files have been
+    written, since dracut depends on some of them.
+
+    :param sysroot: a path to the root of the installed system
+    :param kernel_versions: a list of kernel versions
+    """
+    if os.path.exists(sysroot + "/usr/sbin/new-kernel-pkg"):
+        use_dracut = False
+    else:
+        log.debug("new-kernel-pkg does not exist, using dracut instead")
+        use_dracut = True
+
+    for kernel in kernel_versions:
+        log.info("Recreating initrd for %s", kernel)
+
+        if conf.target.is_image:
+            # hostonly is not sensible for disk image installations
+            # using /dev/disk/by-uuid/ is necessary due to disk image naming
+            execWithRedirect(
+                "dracut", [
+                    "-N", "--persistent-policy", "by-uuid",
+                    "-f", "/boot/initramfs-%s.img" % kernel, kernel
+                ],
+                root=sysroot
+            )
+        else:
+            if use_dracut:
+                execWithRedirect(
+                    "depmod", ["-a", kernel], root=sysroot
```

Comment:
That would be nice, but I don't know what it does.

Reply:
Hmm, I wonder if @jkonecny12 has an idea what is it doing at this stage (as IIRC kernel module resolving is also part of driver disk machinery).

Reply:
Yes, it resolves dependencies. I understand it the way that we need this because Dracut need to have modules ready to install them to initrd. Reason why this is not solved by scriptlets is that not all modules are used in stage2 but some are only usable and installed to initrd which may not be RPM transaction. However, this is just my assumption.